spring-cloud-stream-binders/spring-cloud-stream-binder-spi/src/main/java/org/springframework/cloud/stream/binder/MessageChannelBinderSupport.java

```diff
@@ -377,10 +378,10 @@ public void setDefaultDurableSubscription(boolean defaultDurableSubscription) {
 	@Override
 	public void afterPropertiesSet() throws Exception {
 		Assert.notNull(this.applicationContext, "The 'applicationContext' property cannot be null");
-		onInit();
 		if (this.evaluationContext == null) {
-			this.evaluationContext = IntegrationContextUtils.getEvaluationContext(getBeanFactory());
+			this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
 		}
+		onInit();
```

Comment:
Yes, makes sense, too.
Having all this `MessageChannelBinderSupport` hierarchy as an internal stuff, there is no so much worry about end-users who won't call `super.afterPropertiesSet()` from his impl.